### ISSUE TITLE
Seems to miss files for onedrive --synchronize --resync --verbose

### ISSUE THREAD

**Issue by Clear2Go** [abraunegg/onedrive#1163]
### Bug Report Details ###
**Describe the bug**

I run onedrive for linux connected to my onedrive for business.  I have a sync_list, however there are 10000 of files.  Onedrive is not running anywhere else on system, or in monitor mode.  
It syncs and takes almost 20h.  It completes, then I run the same command again, it finds more files and syncs them.  I then run it again, and it finds more files and syncs them.  It is important to note the cloud OneDrive is not changing, I have them locked.  It is as if it skips or can not do it all at once, but does not report that.  It simply runs as if it completed. 

**Application and Operating System Details:**
*   Provide your OS & version (CentOS 6.x, Ubuntu 18.x etc) and the output of: `uname -a`
```
mike@Brushpass:~$ uname -a
Linux Brushpass 5.4.0-53-generic #59-Ubuntu SMP Wed Oct 21 09:38:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
*   Are you using a headless system (no gui) or with a gui installed?

GUI is available, but not enabled. This is all cmd line / bash

*   OneDrive Account Type
Business
*   Did you build from source or install from a package?
```
dpkg --list | grep -i onedrive
ii  onedrive                                   2.4.7-0~202011102339~ubuntu20.04.1  amd64        folder synchronization with OneDrive
```

*   OneDrive Application Version: Output of `onedrive --version`

see above

*   OneDrive Application Configuration: Output of `onedrive --display-config`
```
mike@Brushpass:~$ onedrive --display-config
onedrive version                       = v2.4.7-0~202011102339~ubuntu20.04.1
Config path                            = /home/mike/.config/onedrive
Config file found in config path       = false
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /home/mike/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = true
sync_list contents:
/bkup/Glacier/20141109-doreen.tar.gz
/0. Evanta
/Attachments
/Books
/Funny Quotes
/fyeo2
/London
/Manuals
/MyPics
/Sandvine
/Sounds
/Travel
/Warranties
/Whoop
/ZJG
Business Shared Folders configured     = false
```
*   Provide the version of curl you are using: Output of `curl --version`
```
mike@Brushpass:~$ curl --version
curl 7.68.0 (x86_64-pc-linux-gnu) libcurl/7.68.0 OpenSSL/1.1.1f zlib/1.2.11 brotli/1.0.7 libidn2/2.2.0 libpsl/0.21.0 (+libidn2/2.2.0) libssh/0.9.3/openssl/zlib nghttp2/1.40.0 librtmp/2.3
Release-Date: 2020-01-08
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS brotli GSS-API HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL TLS-SRP UnixSockets
```
*   Is your configured 'sync_dir' a local directory or a network mount point?

It is to a local USB storage device

*   What partition format type does your configured 'sync_dir' reside on? Output of: `lsblk -f` 
```
db                                                                                               
├─sdb1                                                                                            
└─sdb2                ntfs        P4TbDisk2 9C54BFDC54BFB77C                          3.4T     8% /media/mike/P4TbDisk2
```

*   Explain your entire configuration setup - is the OneDrive folder shared with any other system, shared with any other platform at the same time, is the OneDrive account you use shared across multiple systems / platforms / Operating Systems and in use at the same time

No. It is not shared and both Cloud and my local are only controlled accessed by me personally

**Comment by abraunegg** [comment]
@Clear2Go 

An issue like you are describing should have some details behind it when the application exits, however because you are saying that there is no clear reason, the reason for this is:
* Some sort of flaky Internet connection somewhere between you and the OneDrive service
* Some sort of 'broken' HTTPS transparent inspection service inspecting your traffic somewhere between you and the OneDrive service

Follow the directions listed here: https://github.com/abraunegg/onedrive/wiki/Debugging-SSL-TLS-Issues-when-the-client-stops-syncing-with-no-crash to provide more details around your issue.

**Comment by Clear2Go** [comment]
I guess I would have expected that the takeaway would be "hmmm, how do we address this in the software?".  While I can do this, I really assumed the takeaway would be to consider adding this so that the software will improve with less technical know how so that people that are not technical could use it.  I know many that while they would love to sync their cloud data, are not going to know how or care to take the time to get into ssl/tls debug details. 

In short it was a suggestion .... I'll close this.

**Comment by abraunegg** [comment]
> I guess I would have expected that the takeaway would be "hmmm, how do we address this in the software?". While I can do this, I really assumed the takeaway would be to consider adding this so that the software will improve with less technical know how so that people that are not technical could use it. I know many that while they would love to sync their cloud data, are not going to know how or care to take the time to get into ssl/tls debug details.
> 

If curl exits (which this client is dependent on), without throwing an error or generating an error message unless you are in a debug situation, calling applications have no visibility that the underlying dependent component has failed.

Over the years I have *tried* to solve this in the code, to be as robust as possible - but this is one where - and I am more than happy to be proved wrong here - nothing can be done. The client is highly resilient in 99% of all use cases and re-tries the previous action.

I get your frustration, I get that this is *not* the answer you were looking for - but there is *very* limited fixing this in *this* codebase that can be done to resolve this situation.

**Comment by Clear2Go** [comment]
Do understand. Personally, I am suspicious of the TLS/SSL issue because I can simultaneously with another system execute a large transfer, and it works without session timing out, but there could be other reasons for that that I have not diagnosed.

**Comment by abraunegg** [comment]
@Clear2Go 
Please can you generate the HTTPS Debug Log if you can keep on replicating this issue. This will give you the information points that detail what is causing your error. I have *never* been able to replicate this specific issue - despite months long testing, burning through ~500GB of upload / download data, transferring 1M files on constant repeat / refresh to try and trigger this issue.

Additionally, ensure you are also using the latest 'master' codebase as the Ubuntu package contains other issues, such as compiler library bugs. You are *always* better off self compiling because of this Ubuntu package shenanigans.
